components/director/internal/open_resource_discovery/validation_test.go

```diff
@@ -1493,6 +1519,86 @@ func TestDocuments_ValidateAPI(t *testing.T) {
+				doc := fixORDDocument()
+				doc.APIResources[0].ResourceDefinitions[0].URL = "http://newurl.com/odata/$metadata"
+				doc.APIResources[0].VersionInput.Value = fixAPIsWithLowerVersion()[0].Version.Value
+
+				newHash, err := open_resource_discovery.HashObject(doc.APIResources[0])
+				require.NoError(t, err)
+
+				resourceHashes[api1ORDID] = newHash
+
+				return []*open_resource_discovery.Document{doc}
+			},
+			AfterTest: func() {
+				resourceHashes = fixResourceHashes()
+			},
+		}, {
+			Name: "Not incremented `version` field when resource definition's MediaType has changed for API",
+			DocumentProvider: func() []*open_resource_discovery.Document {
+				doc := fixORDDocument()
+				doc.APIResources[0].ResourceDefinitions[0].MediaType = model.SpecFormatTextYAML
+				doc.APIResources[0].VersionInput.Value = fixAPIsWithLowerVersion()[0].Version.Value
+
+				newHash, err := open_resource_discovery.HashObject(doc.APIResources[0])
+				require.NoError(t, err)
+
+				resourceHashes[api1ORDID] = newHash
+
+				return []*open_resource_discovery.Document{doc}
+			},
+			AfterTest: func() {
+				resourceHashes = fixResourceHashes()
+			},
+		}, {
+			Name: "Not incremented `version` field when resource definition's Type has changed for API",
+			DocumentProvider: func() []*open_resource_discovery.Document {
+				doc := fixORDDocument()
+				doc.APIResources[0].ResourceDefinitions[0].Type = model.APISpecTypeOpenAPIV2
+				doc.APIResources[0].VersionInput.Value = fixAPIsWithLowerVersion()[0].Version.Value
+
+				newHash, err := open_resource_discovery.HashObject(doc.APIResources[0])
+				require.NoError(t, err)
+
+				resourceHashes[api1ORDID] = newHash
+
+				return []*open_resource_discovery.Document{doc}
+			},
+			AfterTest: func() {
+				resourceHashes = fixResourceHashes()
+			},
+		}, {
+			Name: "Not incremented `version` field when resource definition's CustomType has changed for API",
+			DocumentProvider: func() []*open_resource_discovery.Document {
+				doc := fixORDDocument()
+				doc.APIResources[0].ResourceDefinitions[0].Type = model.APISpecTypeCustom
+				doc.APIResources[0].ResourceDefinitions[0].CustomType = "sap:custom-definition-format:v1"
+				doc.APIResources[0].VersionInput.Value = fixAPIsWithLowerVersion()[0].Version.Value
+
+				newHash, err := open_resource_discovery.HashObject(doc.APIResources[0])
+				require.NoError(t, err)
+
+				resourceHashes[api1ORDID] = newHash
+
+				return []*open_resource_discovery.Document{doc}
+			},
+			AfterTest: func() {
+				resourceHashes = fixResourceHashes()
+			},
+		}, {
+			Name: "Valid incremented `version` field when resource definition has changed for API",
+			DocumentProvider: func() []*open_resource_discovery.Document {
+				doc := fixORDDocument()
+				doc.APIResources[0].ResourceDefinitions[0].Type = model.APISpecTypeCustom
+				doc.APIResources[0].ResourceDefinitions[0].CustomType = "sap:custom-definition-format:v1"
+				doc.APIResources[0].VersionInput.Value = "2.1.3"
```

Comment:
In all positive test cases for the version bump, it is pointless to just set a bumped version to the given Resource without re-evaluating the hash.

As mentioned in the `fixtures_test.go` comment, the hash comparison is made between the two fixed objects. If you don't override the hash value for the given changed resource in the `resourceHashes` map (and then reset it in the AfterTest func) then the comparison will never "find out" that there was a change.

To sum up, after a change is made and a correct version bump is set, the hash needs to be re-evaluated and the hash map to be overridden. By doing so, you will simulate actual change in the input and the comparison will be made with the changed input object.